oak_functions/sdk/oak_functions/src/lib.rs

```diff
@@ -90,6 +92,76 @@ pub fn write_response(buf: &[u8]) -> Result<(), OakStatus> {
     result_from_status(status as i32, ())
 }
 
+/// Looks up an item from the in-memory lookup store.
+///
+/// If an entry is not found, returns `Ok(None)` for convenience, instead of
+/// `Err(OakStatus::ErrStorageItemNotFound)`.
+///
+/// See [`storage_get_item`](https://github.com/project-oak/oak/blob/main/docs/oak_functions_abi.md#storage_get_item).
+pub fn storage_get_item(key: &[u8]) -> Result<Option<Vec<u8>>, OakStatus> {
+    // TODO(#1989): Share this logic with other similar methods.
+
+    // It is possible that the item for a given key changes or gets deleted entirely between the two
```

Comment:
I'm not sure there is anything to actually do, this behaviour is documented in the ABI so that's the contract, I think it's fine to leave as it is. #754 may solve it, but I don't think we have decided for it yet. Anyways, let me add the TODO to keep track of the reference.

Reply:
I think this is a serious bug. But this can fixed later.